### DISCUSSION TITLE
Edgeware local development environment setup

### DISCUSSION THREAD

**Discussion by nymkappa** [icon-project/btp#59] [Q&A] [answered]
This question is mostly addressed to ICONDAO team @flowersinthesand @quocle108 but it can be helpful for later reference.

I'm trying to setup a development environment to deploy and test your BTP contracts (`icondao` branch).

So far I have an edgeware local chain running (using `drew-erup-4` branch, running using `docker-compose`). I can see blocks going through on PolkadotUI.

I've setup my Metamask to connect to my local chain, and used Remix to deploy a hello world contract using the injected Web3 provider. I was able to make a couple of transactions, but sometimes, transactions hangs (ex: `creation of HelloWorld pending...`) and my docker CPU consumption goes through the roof. It seems it's related to fee estimation but I'm not 100% sure about this.

Could you describe your development tool stack, how you are running your edgeware local network and if possible could you provide some resources on how to deploy your BTP smart contracts so that we can do some testing on our end as well?

Thanks a lot.

**Comment by trmaphi** [comment]
@jorisvial We can't even compile the source code to build Docker image.  
```
  Compiling cfg-if v0.1.10
   Compiling either v1.6.1
   Compiling erased-serde v0.3.15
   Compiling ref-cast v1.0.6
   Compiling arrayvec v0.4.12
   Compiling tiny-keccak v2.0.2
   Compiling slog v2.7.0
   Compiling bytes v1.0.1
   Compiling pin-project-lite v0.2.6
   Compiling async-trait v0.1.50
   Compiling zstd-safe v3.0.1+zstd.1.4.9
   Compiling futures-io v0.3.15
   Compiling convert_case v0.4.0
   Compiling futures-sink v0.3.15
   Compiling pin-utils v0.1.0
   Compiling subtle v2.4.0
   Compiling opaque-debug v0.3.0
   Compiling static_assertions v1.1.0
error[E0554]: `#![feature]` may not be used on the stable release channel
  --> /Users/trmaphi/.cargo/registry/src/github.com-1ecc6299db9ec823/subtle-2.4.0/src/lib.rs:12:34
   |
12 | #![cfg_attr(feature = "nightly", feature(external_doc))]
   |                                  ^^^^^^^^^^^^^^^^^^^^^
   Compiling tap v1.0.1
error: aborting due to previous error
For more information about this error, try `rustc --explain E0554`.
   Compiling wyz v0.2.0
error: could not compile `subtle`
To learn more, run the command again with --verbose.
warning: build failed, waiting for other jobs to finish...
error: build failed
```

cc/ @hir0min

**Reply by hir0min** [reply]
@jorisvial I got this error when building Docker image
```
executor failed running [/bin/sh -c curl https://raw.githubusercontent.com/rust-lang/rustup/master/rustup-init.sh -sSf | sh -s -- -y &&     export PATH=$PATH:$HOME/.cargo/bin &&     cd source &&     git submodule update --init --recursive &&     rustup default stable &&     rustup uninstall nightly &&     rustup toolchain install ${RUST_VERSION} &&     rustup target add wasm32-unknown-unknown --toolchain ${RUST_VERSION} &&     cargo build -p ${PACKAGE_NAME} --${PROFILE} &&     cp target/${PROFILE}/${BINARY_NAME} /node]: exit code: 2
```

**Reply by nymkappa** [reply]
I just used docker based on this https://github.com/hicommonwealth/edgeware-contracts-tutorials/blob/master/4/setting-up-a-local-node.md

There are more docs here https://github.com/hicommonwealth/edgeware-contracts-tutorials but I found out that many links are dead so it's a bit messy overall

**Reply by hir0min** [reply]
@jorisvial Docker compose uses image `hicommonwealth/edgeware:v3.2.0`, which doesn't have the pre-compiles. Could you update that image to the latest version on branch `drew-erup-4`?

**Reply by nymkappa** [reply]
Good catch. I'm trying to build the image as well but I'm getting stuck too. I've created an issue on their github, let's see how it goes. In the mean time please let us know if you have any luck!

https://github.com/hicommonwealth/edgeware-node/issues/307

**Comment by nymkappa** [comment, marked as answer]
Since you will focus your effort on Polkadot parachain I will close this issue for now.
